cmd/qsctl/presign.go

```diff
@@ -24,20 +32,59 @@ this URL can always retrieve the object with an HTTP GET request.`,
 	PreRun: validatePresignFlag,
 }
 
+func presignParse(t *task.PresignTask, args []string) (err error) {
+	// Parse flags.
+	t.SetExpire(presignInput.expire)
+	return nil
+}
+
 func presignRun(_ *cobra.Command, args []string) error {
+	t := task.NewPresignTask(func(t *task.PresignTask) {
+		if err := presignParse(t, args); err != nil {
+			panic(err)
+		}
+
+		keyType, bucketName, objectKey, err := utils.ParseKey(args[0])
+		if err != nil {
+			panic(err)
+		}
+		// only handle object key, if dir, it's meaningless
+		if keyType != constants.KeyTypeObject {
+			panic(constants.ErrorQsPathInvalid)
+		}
+		t.SetBucketName(bucketName)
+		t.SetKey(objectKey)
+
+		stor, err := storage.NewQingStorObjectStorage()
+		if err != nil {
+			return
+		}
+		t.SetStorage(stor)
+
+		err = stor.SetupBucket(bucketName, "")
+		if err != nil {
+			return
+		}
+
+		// init blank url
+		t.SetURL("")
```

Comment:
We dot't need to init `types`, this call will make `URL` valid check lose effectiveness.